File: 2998-count-symmetric-integers/2998-count-symmetric-integers.java
class Solution {
    public int countSymmetricIntegers(int low, int high) {
        int count = 0;
        for(int i=low;i<=high;i++){
            int num = i;
            if(helper(num)){
                count++;
            }
        }
        return count;
    }
    private boolean helper(int num){
        int temp_num = num;
        int len = 0;
        while(temp_num>0){
            temp_num = temp_num / 10;
            len++;
        }
        if(len%2 != 0){
            return false;
        }
        String str = Integer.toString(num);
        int len_str = str.length();
        int frst = 0;
        int sec = 0;
        for(int i=0;i<len_str/2;i++){
            frst +=  str.charAt(i)-'0';
        }
        for(int j=len_str/2;j<len_str;j++){
            sec += str.charAt(j)-'0';
        }
        if(frst == sec){
            return true;
        }
        return false;
    }
}

File: 2998-count-symmetric-integers/2998-count-symmetric-integers.py
class Solution:
    def countSymmetricIntegers(self, low: int, high: int) -> int:
        count = 0
        for i in range(low,high+1):
            str_num = str(i)
            length = len(str_num)
            if length%2 != 0:
                continue
            if self.helper(str_num):
                count+=1
        
        return count
     
    def helper(self,num:str) -> bool:
        frst = 0
        scd = 0
        len_str = len(num)
        for i in range(0,len_str//2):
            frst += int(num[i])
        
        for j in range(len_str//2,len_str):
            scd += int(num[j])
        
        if (frst == scd):
            return True
        
        return False